e2e/cypress/integration/channel_sidebar/category_collapsing_spec.js

```diff
@@ -166,4 +166,35 @@ describe('Channel sidebar', () => {
         cy.get('.SidebarChannelGroupHeader:contains(CHANNELS) i').should('be.visible').should('have.class', 'icon-rotate-minus-90');
         cy.get('.SidebarChannelGroupHeader:contains(DIRECT MESSAGES) i').should('be.visible').should('not.have.class', 'icon-rotate-minus-90');
     });
+
+    it('should collapse channels that do not have a draft message', () => {
+        cy.getCurrentTeamId().then((teamId) => {
+            cy.apiCreateChannel(teamId, 'channel-test', 'Channel Test').then(() => {
+                // # Create the draft message
+                const draft = `Draft message ${getRandomId()}`;
+                cy.get('#post_textbox').clear().type(draft);
+
+                // # Check that the CHANNELS group header is visible
+                cy.get('.SidebarChannelGroupHeader:contains(CHANNELS)').should('be.visible').as('channelsGroup');
+
+                // * Verify that all channels are visible
+                cy.get('.SidebarChannel:contains(Off-Topic)').should('be.visible');
+                cy.get('.SidebarChannel:contains(Town Square)').should('be.visible');
+                cy.get('.SidebarChannel:contains(Channel Test)').should('be.visible');
+
+                // # Open Off Topic
+                cy.get('.SidebarChannel:contains(Off-Topic)').click();
+
+                // # Click on CHANNELS
+                cy.get('@channelsGroup').click();
+
+                // * Verify that Off-Topic and Town-Square are visible
```

Comment:
I think this step needs to be changed to "Verify that Off-Topic and Channel Test are visible" after we collapse Channels category (line 189)
Followed by "Verify that Town Square is not visible" since it is neither currently open, unread nor containing a draft message.